src/sagemaker_xgboost_container/algorithm_mode/hyperparameter_validation.py

```diff
@@ -187,8 +186,8 @@ def eval_metric_dep_validator(value, dependencies):
         hpv.CategoricalHyperparameter(name="objective",
                                       range=["binary:logistic", "binary:logitraw", "binary:hinge",
                                              "count:poisson", "multi:softmax", "multi:softprob",
```

Comment:
I have some doubt about allowing multi-class classification objective when the AUC metric is set. To my knowledge, Area Under the [ROC Curve](https://en.wikipedia.org/wiki/Receiver_operating_characteristic) is only defined when the number of classes is 2. To plot ROC, we need to convert the multi-class classification problem into one-vs-all. See discussion at https://stats.stackexchange.com/questions/2151/how-to-plot-roc-curves-in-multiclass-classification

Reply:
Let me do some write-up about the definition of AUC